dm/syncer/data_validator.go

```diff
@@ -628,15 +629,39 @@ func (v *DataValidator) doValidate() {
 		default:
 			currLoc.Position.Pos = e.Header.LogPos
 		}
-
+		// update validator metric
+		metrics.ValidatorBinlogPos.WithLabelValues(v.cfg.Name, v.cfg.SourceID, v.cfg.WorkerName).Set(float64(currLoc.Position.Pos))
+		index, err := binlog.GetFilenameIndex(currLoc.Position.Name)
+		if err != nil {
+			v.L.Warn("fail to record validator binlog file index")
+		} else {
+			metrics.ValidatorBinlogFile.WithLabelValues(v.cfg.Name, v.cfg.SourceID, v.cfg.WorkerName).Set(float64(index))
+		}
 		// wait until syncer synced current event
 		err = v.waitSyncerSynced(currLoc)
 		if err != nil {
 			// no need to wrap it in error_list, since err can be context.Canceled only.
 			v.sendError(err)
 			return
 		}
-
+		v.syncer.currentLocationMu.RLock()
```

Comment:
move all those metrics stuff(previous part too) into a separate method to avoid messing the main logic up.